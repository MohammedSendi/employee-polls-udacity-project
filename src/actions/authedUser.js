import {checkUserCredenrials} from '../utils/API'
import {showLoading, hideLoading} from "react-redux-loading-bar"

export const SET_AUTHED_USER = "SET_AUTHED_USER"
export const LOGOUT_USER = "LOGOUT_USER"

export function logoutUser(){
    return {
        type: LOGOUT_USER
    }
}

function setAuthedUser(id) {
    return {
        type: SET_AUTHED_USER,
        id,
    };
}

export function handleLogin(id, password){
    return (dispatch) => {
        
        dispatch(showLoading());
        return new Promise(() => { 
            checkUserCredenrials(id, password).then(user => {
                dispatch(setAuthedUser(user.id))
                dispatch(hideLoading())
            }).catch(() => dispatch(hideLoading()))

        })



    }
}