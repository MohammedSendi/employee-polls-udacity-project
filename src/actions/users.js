export const RECEIVE_USERS = "RECEIVE_USERS"
export const ADD_QUESTION = "ADD_QUESTION"
export const ADD_ANSWER = "ADD_ANSWER"

export function receiveUsers (users) {
    return {
        type: RECEIVE_USERS,
        users,
    }
}

export function addQuestion(id, question){
    return {
        type: ADD_QUESTION,
        id,
        question,
    }
}

export function addAnswer(answer){
    console.log(answer.authedUser)
    return {
        type: ADD_ANSWER,
        id : answer.authedUser,
        qid : answer.qid,
        answer : answer.answer
    }
}