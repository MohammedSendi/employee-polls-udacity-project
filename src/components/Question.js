import { connect } from "react-redux";
import { formatDate } from "../utils/helper";
import { Link } from 'react-router-dom'
import * as React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import { CardActionArea } from '@mui/material';
import Divider from '@mui/material/Divider';

const Question = (props) => {

    return (
      <Card >
        <CardActionArea>
          <Link to={`/poll/${props.question.id}`} style={{ textDecoration: 'none' }}>
          <CardContent >
            <Typography gutterBottom variant="h5" component="div" color="text.primary" textAlign='center'>
              {props.author}
            </Typography>
            <Divider />
            <Typography variant="body2" color="text.secondary" textAlign='center'>
              {formatDate(props.question.timestamp)}
            </Typography>
          </CardContent>
          </Link>
        </CardActionArea>
      </Card>
    )
}

const mapStateToProps = ({authedUser, users, questions}, {id}) => {
     const question = questions[id];
     
     return {
        authedUser,
        question,
        author: users[question.author].name,
     }
}

export default connect(mapStateToProps)(Question)