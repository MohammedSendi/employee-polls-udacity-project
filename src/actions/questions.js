import {_saveQuestion, _saveQuestionAnswer} from '../utils/_DATA'
import { showLoading, hideLoading } from 'react-redux-loading-bar';
import {addQuestion , addAnswer} from './users'

export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const SAVE_QUESTION = "SAVE_QUESTION"
export const SAVE_ANSWER = "SAVE_ANSWER"

export function receiveQuestions(questions){
    return {
        type: RECEIVE_QUESTIONS,
        questions
    }
}

function saveQuestion(question){
    return {
        type: SAVE_QUESTION,
        question,
    }
}

function saveAnswer(answer){
    return {
        type: SAVE_ANSWER,
        answer
    }
}

export function handleSaveQuestion(question){
    return (dispatch) => {
        dispatch(showLoading());
        return _saveQuestion(question)
        .then((question) => {
            dispatch(saveQuestion(question))
            dispatch(addQuestion(question.author, question.id))
        })
        .then(() => dispatch(hideLoading()))
    }
}


export function handleSaveAnswer(answer){
    return (dispatch) => {
        dispatch(showLoading());
        return _saveQuestionAnswer(answer)
        .then(dispatch(saveAnswer(answer)))
        .then(dispatch(addAnswer(answer)))
        .then(() => dispatch(hideLoading()))
    }
}