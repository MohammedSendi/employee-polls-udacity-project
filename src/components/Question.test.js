import { render, fireEvent } from "@testing-library/react";
import * as React from 'react'
import Question from './Question'
import store from "../app/store"
import { Provider } from 'react-redux';
import {handleInitialData} from '../actions/shared'
import {Route, Routes, MemoryRouter} from "react-router-dom"

describe('Question', () => {
    it("matches the snapshot when an id is passed", async() => {
        await store.dispatch(handleInitialData())
        var component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Question id={'vthrdm985a262al8qx3do'}/>            
                </Provider>
            </MemoryRouter>
        )
        expect(component).toMatchSnapshot();

    })
})