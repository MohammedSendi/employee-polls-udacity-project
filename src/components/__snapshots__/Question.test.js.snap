// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question matches the snapshot when an id is passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <a
            href="/questions/vthrdm985a262al8qx3do"
            style="text-decoration: none;"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1hczlon-MuiTypography-root"
              >
                Tyler McGinnis
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body2 css-1da0jfs-MuiTypography-root"
              >
                15:9 | 15/3/2017
              </p>
            </div>
          </a>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <a
          href="/questions/vthrdm985a262al8qx3do"
          style="text-decoration: none;"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1hczlon-MuiTypography-root"
            >
              Tyler McGinnis
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <p
              class="MuiTypography-root MuiTypography-body2 css-1da0jfs-MuiTypography-root"
            >
              15:9 | 15/3/2017
            </p>
          </div>
        </a>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
